korman/exporter/mesh.py

```diff
@@ -265,10 +298,59 @@ def _export_geometry(self, bo, mesh, materials, geospans):
             if numVerts > _WARN_VERTS_PER_SPAN:
                 raise explosions.TooManyVerticesError(bo.data.name, geospan.material.name, numVerts)
 
+            # If we're bump mapping, we need to normalize our magic UVW channels
+            if has_bumpmap is not None:
+                geospan.format += 2 # We dded 2 special bumpmapping UV layers
+                for v, vtx in enumerate(data.vertices):
+                    uvMap = vtx.uvs
+                    uvMap[geospan.format - 2].normalize()
+                    uvMap[geospan.format - 1].normalize()
+                    vtx.uvs = uvMap
+
             # If we're still here, let's add our data to the GeometrySpan
             geospan.indices = data.triangles
             geospan.vertices = data.vertices
 
+
+    def _get_bump_gradient(self, xform, uvws, mesh, vIds, uvIdx, iUV):
+        v0 = hsVector3(*mesh.vertices[vIds[0]].co)
+        v1 = hsVector3(*mesh.vertices[vIds[1]].co)
+        v2 = hsVector3(*mesh.vertices[vIds[2]].co)
+
+        uv0 = [(uv[0], uv[1], 0.0) for uv in uvws[0]][uvIdx]
+        uv1 = [(uv[0], uv[1], 0.0) for uv in uvws[1]][uvIdx]
+        uv2 = [(uv[0], uv[1], 0.0) for uv in uvws[2]][uvIdx]
```

Comment:
Wait. If we know which indices we want, why do we need to use a list comprehension?